scrapy/utils/display.py

```diff
@@ -3,20 +3,63 @@
 """
 
 import sys
+from packaging.version import parse
+from platform import version
 from pprint import pformat as pformat_
 
 
 def _colorize(text, colorize=True):
     if not colorize or not sys.stdout.isatty():
         return text
     try:
+        import pygments.util
+        from pygments.formatters import get_formatter_by_name
         from pygments import highlight
-        from pygments.formatters import TerminalFormatter
         from pygments.lexers import PythonLexer
-        return highlight(text, PythonLexer(), TerminalFormatter())
     except ImportError:
         return text
 
+    # All Windows versions >= "10.0.14393" interpret ANSI escape sequences
+    # using terminal processing.
+    #
+    # Enable enivornment variable `ENABLE_VIRTUAL_TERMINAL_PROCESSING`
+    # to activate terminal processing.
+    if sys.platform == "win32" and parse(version()) >= parse("10.0.14393"):
+        try:
+            import ctypes
+            from ctypes import FormatError
+            kernel32 = ctypes.windll.kernel32
+            handle = kernel32.GetStdHandle(-11)
+            # set `ENABLE_VIRTUAL_TERMINAL_PROCESSING` flag
+            if not kernel32.SetConsoleMode(handle, 7):
+                raise ValueError
+        except ValueError:
+            sys.stderr.write(FormatError(kernel32.GetLastError()))
+            return text
```

Comment:
I am not sure if there is need of error message `sys.stderr.write(FormatError(kernel32.GetLastError()))`, Maybe we can just return the plain text without the error message. 
@wRAR please guide me here.

Reply:
@Gallaecio Can you please tell me if we need error message 'sys.stderr.write(FormatError(kernel32.GetLastError()))' as the text "The handle is invalid" doesn't look aesthetically good(like in below screenshot)
![1](https://user-images.githubusercontent.com/42249933/79500709-81b0bc00-804a-11ea-8f47-30c58ade78d0.PNG)
)
I think we should just return the plain text when exception is raised.

Reply:
I agree, we should just return the text.